src/overlays/actors/ovl_En_Si/z_en_si.c

```diff
@@ -25,19 +41,108 @@ const ActorInit En_Si_InitVars = {
     (ActorFunc)EnSi_Update,
     (ActorFunc)EnSi_Draw,
 };
-*/
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Si/EnSi_Init.s")
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Si/EnSi_Destroy.s")
+void EnSi_Init(Actor* thisx, GlobalContext* globalCtx) {
+    EnSi* this = THIS;
+
+    Collider_InitCylinder(globalCtx, &this->collider);
+    Collider_SetCylinder(globalCtx, &this->collider, &this->actor, &sCylinderInit);
+    func_80061EFC(&this->actor.colChkInfo, NULL, &D_80AFBADC);
+    Actor_SetScale(&this->actor, 0.025f);
+    this->unk_19C = 0;
+    this->actionFunc = func_80AFB768;
+    this->actor.shape.unk_08 = 42.0f;
+}
+
+void EnSi_Destroy(Actor* thisx, GlobalContext* globalCtx) {
+    EnSi* this = THIS;
+
+    Collider_DestroyCylinder(globalCtx, &this->collider);
+}
+
+s32 func_80AFB748(EnSi* this, GlobalContext* globalCtx) {
+    if (this->collider.base.acFlags & 0x2) {
+        this->collider.base.acFlags &= ~0x2;
+    }
+    return 0;
+}
+
+void func_80AFB768(EnSi* this, GlobalContext* globalCtx) {
+    Player* player = PLAYER;
+
+    if ((this->actor.flags & 0x2000) == 0x2000) {
+        this->actionFunc = func_80AFB89C;
+    } else {
+        Math_SmoothScaleMaxMinF(&this->actor.scale.x, 0.25f, 0.4f, 1.0f, 0.0f);
+        Actor_SetScale(&this->actor, this->actor.scale.x);
+        this->actor.shape.rot.y += 0x400;
+
+        if (func_8008E988(globalCtx) == 0) {
+            func_80AFB748(this, globalCtx);
+
+            if (this->collider.base.maskB & 0x1) {
+                this->collider.base.maskB &= ~0x1;
+                Item_Give(globalCtx, ITEM_SKULL_TOKEN);
+                player->actor.freeze = 0xA;
```

Comment:
This is a frame count so it should be decimal.
```suggestion
                player->actor.freeze = 10;
```